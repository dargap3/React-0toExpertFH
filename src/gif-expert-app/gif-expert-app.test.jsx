import { shallow } from "enzyme";

import GifExpertApp from "./gif-expert-app";


describe('<GifExpertApp /> tests', () => {

  test('should show correctly the component', () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();  
  });

  test('should show a list of categories', () => {

    const categories = ['One Punch', ' Dragon Ball'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
  
});
