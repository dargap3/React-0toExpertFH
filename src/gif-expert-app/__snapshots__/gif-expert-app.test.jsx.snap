// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp /> tests should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
  <GifGrid
    category=" Dragon Ball"
    key=" Dragon Ball"
  />
</Fragment>
`;

exports[`<GifExpertApp /> tests should show correctly the component 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
